src/sign_workflow/sign_args.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import argparse
+import logging
+from pathlib import Path
+from typing import List
+
+
+class SignArgs:
+    ACCEPTED_SIGNATURE_FILE_TYPES = [".sig"]
+
+    target: Path
+    components: List[str]
+    type: str
+    sigtype: str
+    platform: str
+
+    def __init__(self) -> None:
+        parser = argparse.ArgumentParser(description="Sign artifacts")
+        parser.add_argument("target", type=Path, help="Path to local manifest file or artifact directory")
+        parser.add_argument("-c", "--component", type=str, nargs='+', dest="components", help="Component or components to sign")
```

Comment:
@dblock this one still stay at +, not * please change.

Reply:
I haven't changed it.

Reply:
I am a bit confused are you saying this one should stay on + or you are planning to change it later?
Thanks.

Reply:
Oh I thought it was a required parameter.

Reply:
```
signArtifacts(
                            artifactPath: "$WORKSPACE/artifacts",
                            sigtype: SIGNATURE_TYPE,
                            platform: DISTRIBUTION_PLATFORM
                    )
```

For reference we dont need `--component` to be at least 1 to sign.